mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -37,45 +37,74 @@ use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
+    let coin = match lp_coinfind(&ctx, &ticker).await {
         Ok(Some(t)) => t,
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
-    }
-    let (cancelled, still_matching) = try_s!(cancel_orders_by(&ctx, CancelBy::Coin { ticker: ticker.clone() }).await);
-    if !still_matching.is_empty() {
-        let err = json!({
-            "error": format!("There're currently matching orders using {}", ticker),
-            "orders": {
-                "matching": still_matching,
-                "cancelled": cancelled,
-            }
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let platform_ticker = coin.platform_ticker();
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx.get_coins_to_disable(platform_ticker, &ticker).await;
+
+    let mut disabled_tokens_tickers = vec![];
+    let mut cancelled_orders = vec![];
+    for ticker in &coins_to_disable {
+        log!("disabling {ticker} coin");
+        let swaps = try_s!(active_swaps_using_coin(&ctx, ticker));
+        if !swaps.is_empty() {
+            let err = json!({
+                "error": format!("There're active swaps using {}", ticker),
+                "swaps": swaps,
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+        let (cancelled, still_matching) = try_s!(
+            cancel_orders_by(&ctx, CancelBy::Coin {
+                ticker: ticker.to_string()
+            })
+            .await
+        );
+
+        if !still_matching.is_empty() {
+            let err = json!({
+                "error": format!("There're currently matching orders using {}", ticker),
+                "orders": {
+                    "matching": still_matching,
+                    "cancelled": &cancelled,
+                }
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+
+        try_s!(disable_coin_impl(&ctx, ticker).await);
+        // Abort all coin related futures on coin deactivation
+        if let Err(err) = coin.on_disabled() {
```

Comment:
@borngraced @sergeyboyko0791 Don't you think that implementing the `Drop` trait for `MmCoinEnum` or every coin would be better? we wouldn't have to remember calling `on_disabled` in the future in such case.

Reply:
Yes but there're weird behaviours with this method which I think the overheat cost is too much for that achievement! 
.e.g for `LighteningCoin`, after impetmenting the drop trait got multiple errors from diff file in the module 
```rust
 cannot move out of type `lightning::LightningCoin`, which implements the `Drop` trait
 
 mm2src/coins/rpc_command/lightning/connect_to_node.rs:88:35
   |
88 |             .save_nodes_addresses(ln_coin.open_channels_nodes)
   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |                                   |
   |                                   cannot move out of here
   |                                   move occurs because `ln_coin.open_channels_nodes` has type `std::sync::Arc<parking_lot::lock_api::Mutex<parking_lot::RawMutex, std::collections::HashMap<secp256k1::PublicKey, std::net::SocketAddr>>>`, which does not implement the `Copy` trait

error[E0509]: cannot move out of type `lightning::LightningCoin`, which implements the `Drop` trait
   --> mm2src/coins/rpc_command/lightning/open_channel.rs:226:31
    |
226 |         .save_nodes_addresses(ln_coin.open_channels_nodes)
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |                               |
    |                               cannot move out of here
    |                               move occurs because `ln_coin.open_channels_nodes` has type `std::sync::Arc<parking_lot::lock_api::Mutex<parking_lot::RawMutex, std::collections::HashMap<secp256k1::PublicKey, std::net::SocketAddr>>>`, which does not implement the `Copy` trait

error[E0509]: cannot move out of type `lightning::LightningCoin`, which implements the `Drop` trait
  --> mm2src/coins/rpc_command/lightning/trusted_nodes.rs:62:46
   |
62 |         ln_coin.persister.save_trusted_nodes(ln_coin.trusted_nodes).await?;
   |                                              ^^^^^^^^^^^^^^^^^^^^^
   |                                              |
   |                                              cannot move out of here
   |                                              move occurs because `ln_coin.trusted_nodes` has type `std::sync::Arc<parking_lot::lock_api::Mutex<parking_lot::RawMutex, std::collections::HashSet<secp256k1::PublicKey>>>`, which does not implement the `Copy` trait

error[E0509]: cannot move out of type `lightning::LightningCoin`, which implements the `Drop` trait
  --> mm2src/coins/rpc_command/lightning/trusted_nodes.rs:91:46
   |
91 |         ln_coin.persister.save_trusted_nodes(ln_coin.trusted_nodes).await?;
   |                                              ^^^^^^^^^^^^^^^^^^^^^
   |                                              |
   |                                              cannot move out of here
   |                                              move occurs because `ln_coin.trusted_nodes` has type `std::sync::Arc<parking_lot::lock_api::Mutex<parking_lot::RawMutex, std::collections::HashSet<secp256k1::PublicKey>>>`, which does not implement the `Copy` trait

error[E0509]: cannot move out of type `lightning::LightningCoin`, which implements the `Drop` trait
  --> mm2src/coins/rpc_command/lightning/update_channel.rs:66:35
   |
66 |           let mut channel_options = ln_coin
   |  ___________________________________^
67 | |             .conf
68 | |             .channel_options
   | |                            ^
   | |                            |
   | |____________________________cannot move out of here
   |                              move occurs because `ln_coin.conf.channel_options` has type `std::option::Option<lightning::ln_conf::ChannelOptions>`, which does not implement the `Copy` trait


```

Reply:
`MmCoinEnum` is a wrapper over `Arc<Coin>`, so `MmCoinEnum` can be dropped multiple times, but the corresponding `Coin` won't be actually dropped.

Also we can't call the `on_disabled` method within `impl Drop for Coin`, because `Arc<Coin>` may never be dropped without aborting all futures.

So I think we need for `MmCoin::on_disabled` method